deltacat/tests/utils/test_daft.py

```diff
@@ -42,18 +42,51 @@ def test_read_from_s3_single_column_via_column_names(self):
         self.assertEqual(table.num_rows, 100)
 
     def test_read_from_s3_single_column_with_schema(self):
-        schema = pa.schema([("a", pa.int64()), ("b", pa.string())])
+        schema = pa.schema([("a", pa.int8()), ("b", pa.string())])
         pa_read_func_kwargs_provider = ReadKwargsProviderPyArrowSchemaOverride(
             schema=schema
         )
         table = daft_s3_file_to_table(
             self.MVP_PATH,
             content_encoding=ContentEncoding.IDENTITY.value,
             content_type=ContentType.PARQUET.value,
-            include_columns=["b"],
+            include_columns=["a"],
             pa_read_func_kwargs_provider=pa_read_func_kwargs_provider,
         )
-        self.assertEqual(table.schema.names, ["b"])
+        self.assertEqual(table.schema.names, ["a"])
+        self.assertEqual(table.schema.field("a").type, pa.int8())
+        self.assertEqual(table.num_rows, 100)
+
+    def test_read_from_s3_single_column_with_schema_reverse_order(self):
+        schema = pa.schema([("b", pa.string()), ("a", pa.int8())])
+        pa_read_func_kwargs_provider = ReadKwargsProviderPyArrowSchemaOverride(
+            schema=schema
+        )
+        table = daft_s3_file_to_table(
+            self.MVP_PATH,
+            content_encoding=ContentEncoding.IDENTITY.value,
+            content_type=ContentType.PARQUET.value,
+            include_columns=["a", "b"],
+            pa_read_func_kwargs_provider=pa_read_func_kwargs_provider,
+        )
+        self.assertEqual(table.schema.names, ["a", "b"])
+        self.assertEqual(table.schema.field("a").type, pa.int8())
```

Comment:
Yeah, it's safe for now. If there is a way to go back to old daft behavior, that would be awesome. Otherwise, we can keep it as it.

Reply:
Oh! Ok yes I prefer the old Daft behavior as well. To clarify, this will be the behavior:

1. Field names will always have the same order as specified by DeltaCat
2. Field types will always have the same types as specified by DeltaCat
3. Any columns specified by DeltaCat that doesn't exist in the parsed file will be padded with `null` arrays
4. Any columns that exist in the parsed table but not in the DeltaCat-provided schema will just be dropped

Reply:
That sounds right.

Reply:
Changed behavior to match the above comment, and added more tests to verify this behavior as well